aerleon/lib/nacaddr.py

```diff
@@ -83,13 +98,13 @@ def __init__(self, ip_string, comment='', token='', strict=True):
             ip = ip_string
         super().__init__(ip, strict)
 
-    def subnet_of(self, other):
+    def subnet_of(self, other: "IPv4") -> bool:
         """Return True if this network is a subnet of other."""
         if self.version != other.version:
             return False
         return self._is_subnet_of(self, other)
 
-    def supernet_of(self, other):
+    def supernet_of(self, other: Union[IPv4, IPv6]) -> bool:
```

Comment:
You can call `ip.supernet_of()` while mixing IPv4 and IPv6? I know there is IPv4 embedding in IPv6 but is that actually used here?

Reply:
You can but it will always return false. self.version is either 4 or 6.

Reply:
Actually something I am unsure about. `version` is defined in _BaseV4 and _BaseV6 which is eventually what `nacaddr.IPv4` and `nacaddr.IPv6` inherit from. Since this is a method on the nacaddr objects should we use those or should we always go as far down the stack and the method would allow us? I would go with the former for methods in a class and the later for methods outside of classes.

Reply:
I think the most important thing is to be communicative. Use the type signature to express what types this code is designed to operate on. Is nacaddr actually designed to be compatible with other subclasses of _BaseNetwork? If it is, it would be good to select types from the `ipaddress` modules. But if you want nacaddr.IPv[4,6] to mainly interoperate with themselves, use those as the acceptable types.